Add binary sensor class

kk7ds opened the pull request:
This is what I was thinking for a binary sensor class, which gives me this:

![](http://i.imgur.com/r7Xakm0.png)

Requires https://github.com/balloob/home-assistant-polymer/pull/32

kk7ds committed:
Add sensor_class to binary_sensor

This adds a 'sensor_class' property and attribute, which should be either
None or one of several defined SENSOR_CLASSES to indicate contextual
information about what the sensor is measuring.

kk7ds committed:
Make nx584 expose zone types (sensor classes)

With this, plus https://github.com/balloob/home-assistant-polymer/pull/32,
I can have nx584 sensors use a proper icon in the UI.

MartinHjelmare commented:
Is it mostly to support frontend enhancement? I like it.

balloob commented on homeassistant/components/binary_sensor/__init__.py:
```diff
@@ -47,3 +57,14 @@ def state(self):
+    @property
+    def sensor_class(self):
+        """ Returns the class of this sensor, from SENSOR_CASSES. """
+        return 'digital'
```
Wonder if we should make `None` default and not add it as an attribute when not set. The UI will do the default behavior.

kk7ds replied:
Sure, that's fine.

balloob commented:
I like it very much :dolphin: I think it's good to go.

kk7ds commented:
Okay I'll clean it up tomorrow.

kk7ds committed:
Update the sensor classes in the demo binary_sensor

This updates the two demo sensors we have so they show contextual icons
in the UI.
